### DISCUSSION TITLE
Simpler way to enable STARTTLS

### DISCUSSION THREAD

**Discussion by Trapulo** [postalserver/postal#2066] [Help with using Postal]
Hi,
I've a running Postal, that is working behind HAProxy (pfSense) as reverse proxy. HAProxy does SSL offloading, using certificates from Letsencrypt. So HTTPS access to Postal works well.
SMTP is exposed with a simple NAT.

Now I need to enable STARTTLS to protect authentication of clients submitting emails via SMTP. I cannot find a way to perform this. Someone here I see that is installing LE cert bot on host, then is sharing certificates with Postal container. But my host is not exposed to Internet (its the HAProxy that is public), so LE will not validate requests. And also HAProxy is currently directing LE requests to pfSense internal Web server to manage requests uses to get certificate for HAProxy and HTTPS sessions.

So I cannot find a way to protect SMTP sessions. Any idea?

thanks

**Comment by willpower232** [comment]
Postal needs some certificate files in order to enable TLS so you'll have to find some way of providing them.

You could use Lets Encrypt with DNS validation instead of HTTP validation which would bypass the need to forward validation requests.

**Comment by Trapulo** [comment]
Thank you, this can be an idea. Currently my provider does not support required API, so I need to move DNS to go in this direction.
So the solution is to install an acme client on the host, validate hostname with DNS validation, save certificates in config folder shared with Postal container, and enable TLS in config, right?

I'm curious to know how other people with production systems are providing TLS on SMTP...

**Reply by willpower232** [reply]
That sounds right, [certbot has plugins to a few DNS providers](https://eff-certbot.readthedocs.io/en/stable/using.html#dns-plugins) but there are also alternatives like [go-acme/lego](https://github.com/go-acme/lego) which support even more.

Postal always needs the certificate files so the solution wouldn't be much different if at all.

**Reply by Trapulo** [reply]
I moved to Cloudflare my nameservers, and so I was able to complete all using Certbot + Cloudflare API. Now Postal is working accepting STARTTLS in incoming SMTP sessions and I was able to submit emails from Thunderbird.

The only problem I have is that I'm not sure the hook script I created to update certificates and restart Postal after certbot renewal process will work: tests reports error during restart, but restart seems done.
